### PR TITLE
fix(query): fix memory leak for http kill query

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix(query): fix memory leak for http kill query

- Closes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
Ready for review?

**Review by BohuTANG — APPROVED** [review]

File: src/query/service/src/servers/http/v1/http_query_handlers.rs [hunk 1]
```diff
@@ -264,6 +264,7 @@ async fn query_cancel_handler(
         let http_query_manager = HttpQueryManager::instance();
         match http_query_manager.get_query(&query_id).await {
             Some(query) => {
+                query.detach().await;
```
Comment (BohuTANG) [inline src/query/service/src/servers/http/v1/http_query_handlers.rs:267]: Move `query.detach()` to `query.kill()`?

Comment (BohuTANG) [reply src/query/service/src/servers/http/v1/http_query_handlers.rs:267]: https://github.com/datafuselabs/databend/pull/13606
